pyquil/quilatom.py

```diff
@@ -837,28 +838,37 @@ def _contained_mrefs(expression: ExpressionDesignator) -> Set[MemoryReference]:
         return set()
 
 
-@dataclass(eq=True, frozen=True)
-class Frame(QuilAtom):
+class Frame(quil_rs.FrameIdentifier):
     """
     Representation of a frame descriptor.
     """
 
-    qubits: Tuple[Union[Qubit, FormalArgument], ...]
-    """ A tuple of qubits on which the frame exists. """
+    @staticmethod
+    def __new__(cls, qubits: Sequence[Union[int, Qubit, FormalArgument]], name: str) -> "Frame":
```

Comment:
I feel like `name` should be the first parameter here for consistency with the other instructions. Should we add a deprecation warning?

Reply:
I do like that idea, do we want to commit to that for all instructions? How do we best warn users about that future change? So far we've been adding deprecation warnings, but that would be pretty noisy if _all_ instructions had one.